### ISSUE TITLE
EXC_CRASH (SIGABRT) on Namespace SIGNAL, Code 6 Abort trap: 6 when opening 60MB .sql file

### ISSUE THREAD

**Issue by byronrode** [zed-industries/zed#8528]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Right-click on larger (2MB+) .sql file and select Open With -> Zed
File opens for a few seconds and crashes, irrespective of if the app is running or not.

Attached below is the macOS report.

[Problem Report](https://github.com/zed-industries/zed/files/14433210/scratch_34.txt)

### Environment

Zed: v0.123.6 (Zed)
OS: macOS 14.3.1
Memory: 32 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

2024-02-28T13:26:50+02:00 [INFO] 0 unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-02-28T11:26:50.815Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-02-28T11:26:50.815Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-02-28T13:26:50+02:00 [INFO] 0 unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "24813854-2338-44e7-9e55-00962a729f01",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-02-28T13:26:51+02:00 [INFO] reload git repository ".git"
2024-02-28T13:26:51+02:00 [INFO] 0 unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-02-28T11:26:51.269Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-02-28T11:26:51.269Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-02-28T13:26:52+02:00 [INFO] reload git repository ".git"
2024-02-28T13:26:52+02:00 [INFO] add connection to peer
2024-02-28T13:26:52+02:00 [INFO] waiting for server hello
2024-02-28T13:26:52+02:00 [INFO] got server hello
2024-02-28T13:26:52+02:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 346, id: 960032 })
2024-02-28T13:26:52+02:00 [INFO] set status on client 105314: Connected { peer_id: PeerId { owner_id: 346, id: 960032 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }

**Comment by mrnugget** [comment]
cc @maxbrunsfeld @maxdeviant the panic report contains:

```
Thread 2 Crashed::  Dispatch queue: com.apple.root.user-initiated-qos
0   libsystem_kernel.dylib        	       0x18a2fa0dc __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x18a331cc0 pthread_kill + 288
2   libsystem_c.dylib             	       0x18a23da40 abort + 180
3   zed                           	       0x1078f93fc callback__notify_memory_growth + 44
4   zed                           	       0x10697985c _RINvNtNtCshFSawoH58qX_8wasmtime10trampoline4func15array_call_shimNCINvMs6_NtB6_4funcNtB1a_8HostFunc13new_uncheckedNtNtCslcJNU4DcKSp_14wamstime_c_api5store9StoreDataNCNvNtB1S_4func31c_unchecked_callback_to_rust_fn0E0EB1S_ + 148
5   ???                           	       0x1264c0050 ???
6   ???                           	       0x12e728d28 ???
7   ???                           	       0x12e729abc ???
8   ???                           	       0x12e50d640 ???
9   ???                           	       0x12e50dd1c ???
10  ???                           	       0x12e5238d4 ???
11  zed                           	       0x107901e3c wasmtime_setjmp_16_0_0 + 100
12  zed                           	       0x106979348 _RINvNtCshFSawoH58qX_8wasmtime4func27invoke_wasm_and_catch_trapsNtNtCslcJNU4DcKSp_14wamstime_c_api5store9StoreDataNCINvMB2_NtB2_4Func18call_unchecked_rawBZ_E0EB13_ + 416
13  zed                           	       0x10697cbf4 wasmtime_func_call_unchecked + 164
14  zed                           	       0x1078fb35c ts_wasm_store__call + 120
15  zed                           	       0x1078e96ac ts_parser__advance + 3412
16  zed                           	       0x1078e6d80 ts_parser_parse + 952
17  zed                           	       0x105970f30 _RNvNtCsf9LYHVQnZIe_8language10syntax_map10parse_text + 364
18  zed                           	       0x10596be6c _RNvMs1_NtCsf9LYHVQnZIe_8language10syntax_mapNtB5_14SyntaxSnapshot19reparse_with_ranges + 12108
19  zed                           	       0x105937a1c _RNCNvMNtCsf9LYHVQnZIe_8language6bufferNtB4_6Buffer7reparse0B6_ + 452
20  zed                           	       0x1058fd93c _RNvMs1_NtCskkMlY82nSHu_10async_task3rawINtB5_7RawTaskINtNtCs25QQnN8LMgH_4core3pin3PinINtNtCs4NnepYwyJgR_5alloc5boxed3BoxDNtNtNtBU_6future6future6Futurep6OutputNtNtCsf9LYHVQnZIe_8language10syntax_map14SyntaxSnapshotNtNtBU_6marker4SendEL_EEB2x_NCINvMs3_NtCs2mzBxgaCaKa_4gpui8executorNtB41_18BackgroundExecutor14spawn_internalB2x_E0uE3runB2B_ + 604
21  libdispatch.dylib             	       0x18a180910 _dispatch_client_callout + 20
22  libdispatch.dylib             	       0x18a1923cc _dispatch_root_queue_drain + 864
23  libdispatch.dylib             	       0x18a192a04 _dispatch_worker_thread2 + 156
24  libsystem_pthread.dylib       	       0x18a32e0d8 _pthread_wqthread + 228
25  libsystem_pthread.dylib       	       0x18a32ce30 start_wqthread + 8
```

I see a `wasmtime` in there. Wasn't this fixed on `main`?

**Comment by maxbrunsfeld** [comment]
@byronrode Thanks for the report. By any chance, do you have a SQL file that triggers this crash that you could share? I am able to work on a fix without this, but it'd be helpful for validating the fix.

**Comment by byronrode** [comment]
> @byronrode Thanks for the report. By any chance, do you have a SQL file that triggers this crash that you could share? I am able to work on a fix without this, but it'd be helpful for validating the fix.

Apologies for the delayed reply - I can share a file with you (but would prefer to share privately than here in the issue as it might still contain some information not for public - I have tried to remove as much personally identifiable information from the SQL file).

I also tried to see if I could find what caused it (i.e. was it Cyrillic languages or encoding perhaps), but on my side the only crashing files are WordPress SQL files, because other, larger non-WordPress SQL files don't crash at all. 

1. I tested a 20mb non-WP SQL file and it loads up perfectly
2. The shareable file that does crash is 4.7mb
